### DISCUSSION TITLE
"No configured language server supports hover" error on Space+k keyboard shortcut

### DISCUSSION THREAD

**Discussion by rp1231** [helix-editor/helix#10979] [Q&A] [answered]
Whenever I try to view a function definition using the keyboard shortcut Space+k, I get the following error message:
No configured language server supports hover

How do I solve this issue? 
Thanks

**Comment by kirawi** [comment]
Please provide additional information:
- Helix version (hx --version)
- Language
- Output of hx --health <language_name>
- And the content of `:log-open` when running `hx -v` and trying Space-k

**Reply by rp1231** [reply]
Here is the info:
- Helix version: helix 24.3 (2cadec0b)
- Language: Rust
- Output of hx --health rust:
```
Configured language servers:
  ✓ rust-analyzer: C:\Users\lenovo\.cargo\bin\rust-analyzer.exe
Configured debug adapter: lldb-vscode
Binary for debug adapter: 'lldb-vscode' not found in $PATH
Configured formatter: None
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```
- content of `log-open` when running `hx -v` and trying Space-k

[helix.log](https://github.com/user-attachments/files/15879734/helix.log)

**Reply by kirawi** [reply]
`2024-06-18T08:13:17.596 helix_lsp::transport [ERROR] rust-analyzer err <- "error: Unknown binary 'rust-analyzer.exe' in official toolchain 'stable-x86_64-pc-windows-msvc'.\n"`

I've never seen this error before but you might want to follow up with rust-analyzer on this.

**Reply by rp1231** [reply, marked as answer]
running this command solved it for me:
`rustup component add rust-analyzer`
